iec61850serversimulator/src/main/java/com/cgi/iec61850serversimulator/functionclass/DatabaseUtils.java

```diff
@@ -107,16 +112,20 @@ public void checkSchedule(Schedule schedule, int relayNr) {
 
         // Checks if relay can be found with index number. If not found, then it will
         // copy the model's relay to the database.
-        boolean found = this.scheduleRepository.existsByIndexNumberAndRelayId(schedule.getIndexNumber(), relayNr);
+
+        List<RelayEntity> relayEntities = this.relayRepository.findByIndexNumber(relayNr);
+        RelayEntity relayEntity = relayEntities.get(0);
+        long relayId = relayEntity.getId();
+        boolean found = this.scheduleRepository.existsByIndexNumberAndRelayId(schedule.getIndexNumber(), relayId);
 
         if (found) {
-            logger.info("Schedule found in database. Updating model. NOT IMPLEMENTED AS OF NOW.");
-            // this.updateModelSchedule(schedule, relayNr);
+            logger.info("Schedule found in database. Updating model.");
+            this.updateModelSchedule(schedule, relayNr);
         } else {
 
             if (schedule.isEnabled()) {
```

Comment:
Why isn't it necessary to check isEnabled for entries that were found in the DB?

Reply:
Because deletion was not possible at that time. This has now been implemented along with a deletion feature to delete disabled schedules in the DB to save on space.